### PR TITLE
Massive overhaul of Motor Controller

### LINKED ISSUE #53: Motor Controller Overhaul
After talking with @bot190 we came up with the following proposal:

- Remove `PIDSpeedController` and move the one relevant function (`setSetpoint`) to `SmartSpeedController` with a default behavior of doing nothing. Current `setP` etc. will be configured in the robot map instead on the base object.
- Merge `ModSpeedController` modifier capabilities into `SmartSpeedController`.
- Move configuration into the maps themselves, and wrapping with our object is the last step.
- Rename `SmartSpeedController` and its ilk to `SmartMotorController`

Things this doesn't completely solve: how `SwPIDSpeedController` should be implemented, or if it should be implemented at all.

### PR DESCRIPTION
**Pull request by msoucy**
Fixes #53

### CODE REVIEW

File: src/main/java/com/chopshop166/chopshoplib/outputs/SwPIDSpeedController.java [hunk 1]
```diff
@@ -133,90 +100,24 @@ public PIDController getController() {
...
-        motor.pidWrite(output);
-    }
-
-    @Override
-    public IEncoder getEncoder() {
-        return motor.getEncoder();
-    }
-
     /** Calculate the PID value, and set the speed controler to the result. */
     private void calculate() {
```
Comment (bot190) [inline src/main/java/com/chopshop166/chopshoplib/outputs/SwPIDSpeedController.java:122]: This is aliasing `SmartMotorController.calculate()` isn't it?

File: src/main/java/com/chopshop166/chopshoplib/outputs/SwPIDSpeedController.java [hunk 2]
```diff
@@ -133,90 +100,24 @@ public PIDController getController() {
...
-        pid.setD(kd);
-    }
-
-    @Override
-    public void setF(final double kf) {
-        feedforward = kf;
-    }
-
     @Override
     public void setSetpoint(final double setPoint) {
```
Comment (bot190) [inline src/main/java/com/chopshop166/chopshoplib/outputs/SwPIDSpeedController.java:116]: I think this should call `SmartMotorController.calculate()` when setting the setpoint.

File: src/main/java/com/chopshop166/chopshoplib/outputs/SmartMotorController.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        wrapped.disable();
+    }
+
+    @Override
+    public void stopMotor() {
+        wrapped.stopMotor();
+    }
+
+    @Override
+    public void pidWrite(final double output) {
```
Comment (bot190) [inline src/main/java/com/chopshop166/chopshoplib/outputs/SmartMotorController.java:134]: I don't think we need this anymore right?

File: src/main/java/com/chopshop166/chopshoplib/outputs/SmartMotorController.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        wrapped.set(calculate(speed));
+    }
+
+    @Override
+    public double get() {
+        return wrapped.get();
+    }
+
+    @Override
+    public void setInverted(final boolean isInverted) {
```
Comment (bot190) [inline src/main/java/com/chopshop166/chopshoplib/outputs/SmartMotorController.java:114]: I wonder if it's worth adding a comment on this to remind us that we shouldn't be using this in the subsystem. Whether a motor is inverted is really a property of the robot itself.

**Review by bot190 — COMMENTED** [review]
I think we want `PIDSparkMax.Set()` and `PIDSparkMax.setSetpoint()` to call calculate(speed).

**Comment by msoucy** [comment]
Set should, but set setpoint should not - the same modifiers that work with power would not be set up to handle a measurement as input. And there's no real way to inject the modifier system in the middle of the internal PID calculation...

File: src/main/java/com/chopshop166/chopshoplib/outputs/SwPIDSpeedController.java (hunk 1 above)
Comment (msoucy) [reply src/main/java/com/chopshop166/chopshoplib/outputs/SwPIDSpeedController.java:122]: Overloading - different argument types and counts. I agree that the names could be better though - naming is hard.

File: src/main/java/com/chopshop166/chopshoplib/outputs/SmartMotorController.java (hunk 3 above)
Comment (msoucy) [reply src/main/java/com/chopshop166/chopshoplib/outputs/SmartMotorController.java:134]: It was deprecated for removal, but the removal won't happen until 2022 season, so it should stick around until then

**Review by bot190 — APPROVED** [review]

**Pull request merged**